DevelopGitHub/GitHubExtension/GitHubExtension.EntryPoint/FrontEnd/src/client/app/user/user.service.js

```diff
@@ -27,11 +32,50 @@
             return !!$cookies.get('userName');
         }
 
+        function getRepositories() {
+            return $http({
+                method: 'GET',
+                url: API_URL.REPOSITORY
+            }).then( function(responce){
+                RepositoryList = responce.data;
+                logger.info(i18n.REPOSITORIES_LOAD_MESSAGE);
+            });
+        }
+
+        function getCurrentRepository() {
+            return CurrentRepository;
+        }
+
+        function setCurrentRepository(repository) {
+            return $http({
+                method: 'PATCH',
+                dataType: 'json',
+                url: API_URL.CURRENT_REPOSITORY,
+                data: {
+                    Id: repository.id
+                }
+            }).then( function(){
+                CurrentRepository = repository;
+                $state.reload();
+                logger.info(i18n.CHANGE_REPOSITORY_MESSAGE);
+            }, function() {
+                logger.error(i18n.CHANGE_REPOSITORY_ERROR_MESSAGE);
+            });
+        }
+
+        function getRepositoryList() {
+            return RepositoryList;
+        }
+
         return {
             login: login,
             logout: logout,
             userName: userName,
-            isAuthenticated: isAuthenticated
+            isAuthenticated: isAuthenticated,
+            getCurrentRepository: getCurrentRepository,
+            setCurrentRepository: setCurrentRepository,
+            getRepositories: getRepositories,
+            getRepositoryList: getRepositoryList
```

Comment:
do we need 2 methods that do the same logic?
`getRepositories` can just send cached data if any!